### PR TITLE
feat: `SelectAssetPopup`

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
The code all looks good, just have some UX feedback:
- The search input should work with token names _and_ tickers
- When I use the search input, I feel it should deselect the selected token... if you have search input such that the selected token doesn't show, it is still possible to continue which feels strange.

File: packages/desktop/views/dashboard/send-flow/views/SelectTokenView.svelte [hunk 1]
```diff
@@ -1,29 +1,116 @@
...
+    }
 </script>
 
-<select-token-view class="w-full h-full space-y-6 flex flex-auto flex-col flex-shrink-0">
-    <select-token-title>
-        <Text type={TextType.h3} fontWeight={FontWeight.semibold} classes="text-left">Select Token Title</Text>
-    </select-token-title>
-    <select-token-content> Select Token Content </select-token-content>
-    <select-token-buttons class="flex flex-row flex-nowrap w-full space-x-4">
+<select-asset-popup class="w-full h-full space-y-6 flex flex-auto flex-col flex-shrink-0">
```
Comment (nicole-obrien) [inline packages/desktop/views/dashboard/send-flow/views/SelectTokenView.svelte:71]: select-token-view

File: packages/desktop/views/dashboard/send-flow/views/SelectTokenView.svelte [hunk 2]
```diff
@@ -1,29 +1,116 @@
...
+    <Text type={TextType.h3} fontWeight={FontWeight.semibold} classes="text-left">
+        {localize('popups.transaction.selectToken')}
+    </Text>
+    <select-token-buttons class="w-full space-y-4">
```
Comment (nicole-obrien) [inline packages/desktop/views/dashboard/send-flow/views/SelectTokenView.svelte:75]: select-token-content

**Review by nicole-obrien — CHANGES_REQUESTED** [review]

File: packages/desktop/views/dashboard/send-flow/views/SelectTokenView.svelte [hunk 3]
```diff
@@ -1,29 +1,97 @@
...
+
+    function setFilteredAssetList(): void {
+        if (!assets) {
+            assetList = []
+        } else {
+            assetList = [assets.baseCoin, ...assets.nativeTokens].filter(isVisibleAsset)
+        }
+        if (!assetList.some((asset) => asset.id === selectedAssetId)) {
+            selectedAssetId = undefined
+        }
```
[the remark this answers is not in the document]
Comment (MarkNerdi) [reply packages/desktop/views/dashboard/send-flow/views/SelectTokenView.svelte:39]: if we filter the list and the once selected asset isnt included in that filtered list, we want to unset it and force the user to select a new one.

if the selected one is still in it, then we dont want to unselect it

**Pull request merged**